### ISSUE TITLE
study(rl): canonical cadence PPO 归因与 control re-freeze

### ISSUE THREAD

**Issue by xcz0** [xcz0/Eco-AutoDrive#105]
Parent: #80

Blocks: #83（以及后续 formal λ sweep）

## 背景

ADR 0039 已将正式 train/eval execution contract 统一为 canonical k=5 / 0.5 s。#83 的 E-049 复测表明，旧 E-039 effective-update control 在新 cadence 下未通过 Gate T2：

- median seeded-MC post-update KL = 3.258e-7 < 1e-6；
- ratio / probe RMS / parameter delta 均显示 policy 并非 no-op；
- E-040 在旧 k=1 + calibrated R0 下 MC KL=1.809e-6，说明 calibrated R0 本身不是充分解释；
- k=5 同时改变 reward/value scale、gamma/GAE 的物理时间语义、state/action distribution，以及 actor/critic shared-trunk update geometry。

因此需要把 cadence-PPO 归因与 canonical control 重标定从 λ 主效应研究中独立出来，避免把 optimizer/temporal/value-scale 搜索与 λ sweep 混在同一 Issue。

## 目标

回答：

1. E-039 → E-049 的 effective-update 下降主要来自 cadence-induced actor geometry、reward/value scale、temporal credit、critic/shared-trunk coupling，还是其组合；
2. 在 canonical k=5 下重新冻结一个 calibrated R0 的 PPO control；
3. 为后续 transfer / positive-control / λ sweep 提供独立、可核验的 PPO provenance。

## 冻结诊断 contract

Phase A 的 matched attribution 使用：

- 同一 current clean commit；
- calibrated R0；
- 同一 initial policy；
- 同一 training seed / replay / scenario pool；
- batch = 128；
- 50-update budget；
- E-039 candidate 作为 historical comparator；
- k=1 仅作 causal diagnostic，正式 protocol 仍为 k=5。

### Phase A — matched attribution

做：

~~~text
A: k=1 diagnostic execution
B: k=5 canonical execution
~~~

必须保存并比较：

- raw / centered / z-normalized advantage；
- value-target mean/std；
- reward total / component scale；
- GAE / bootstrap statistics；
- gradient_diagnostics=true：
  - actor-head policy gradient；
  - shared-trunk policy gradient；
  - value-head critic gradient；
  - shared-trunk critic gradient；
  - entropy gradient；
- total pre-clip norm、post-clip norm、effective global clip coefficient；
- actor-head / shared-trunk / value-head parameter delta；
- policy ratio；
- Beta mean / concentration / boundary；
- post-update KL：
  - analytic Beta KL；
  - seeded-MC KL；
  - k3 estimate；
- final-policy matched held-out diagnostic，即使 mechanical KL floor 未过也执行一次；该 held-out 只用于归因，不反向改写历史 Gate T2。

### Gate A

进入 counterfactual 前，应能明确至少缩小到以下一项或组合：

- actor/state/advantage geometry；
- reward/value scale；
- physical-time credit semantics；
- critic/shared-trunk/global-clipping coupling。

若证据不足，补最小诊断；不进入 optimizer 搜索。

## Phase B — causal counterfactuals

只在 Phase A 后按证据最小化执行；这些均为 diagnostic arms，不自动成为正式 #83 contract。

优先顺序：

1. **reward-scale ablation**：k=5 state/action cadence 不变，对 sum reward 与 diagnostic scale-normalized reward；
2. **temporal-credit ablation**：当前 transition-based gamma/lambda 与保持旧 0.1 s 物理衰减时间常数的 time-consistent mapping（例如 gamma^5 / lambda^5 形式的诊断臂）；
3. **actor/critic coupling ablation**：value-loss scaling、shared-trunk critic gradient、global clipping；必要时 actor-only / separated clipping 仅作 counterfactual。

### Gate B

Phase B 的结论必须能说明：

- 哪个机制改变 actor effective update；
- 哪些变化只是 critic/value scale 后果；
- 哪些诊断臂不应进入正式 control。

不得把“某个设置 KL 更大”直接等同为正确正式配置。

## Phase C — independent canonical PPO re-freeze

只有 Phase A/B 给出足够归因后才允许重新定位 canonical effective-update region。

允许考虑：

- learning rate；
- epochs / minibatch；
- time-consistent discount / GAE；
- value-loss scaling；
- clipping / actor-critic coupling。

约束：

- 必须使用 calibrated R0；
- 必须保持 canonical k=5 execution；
- 不改变 energy reward representation / λ / policy feature；
- 搜索空间与选择规则须在看到结果前冻结；
- 不按单一 training return 选 control。

新 candidate 至少应验证：

- non-trivial post-update KL / ratio / Beta-mean shift；
- KL 无持续 runaway；
- 无 NaN / Inf / invalid action / Beta boundary collapse；
- collision / OOR / episode-length 无 pathological collapse；
- analytic Beta KL 与 MC/k3 对 effective-update 强度给出一致数量级判断；
- provenance 完整、commit clean。

## 输出 / 完成判据

本 Issue 完成时必须产出：

- [ ] cadence-PPO attribution 结论，明确主要机制与剩余不确定性；
- [ ] analytic Beta KL 或等价可靠 estimator 已落地并验证；
- [ ] 一个冻结的 canonical k=5 PPO control，含 resolved config；
- [ ] 独立实验记录：commit、dirty status、seeds、commands、artifacts、diagnostics；
- [ ] 将新 control 与 provenance 回写到后续 transfer/sweep Issue；
- [ ] 明确哪些 Phase B counterfactual 只是诊断、未进入正式 contract。

## 非目标

本 Issue 不做：

- λ={1,2,4,8} dose-response；
- λ=64 positive-control direction test；
- 候选多 seed 长程确认；
- A0/A1/A2 final matched evaluation；
- reward representation 与 PPO control 的联合搜索。
